packages/discovery-react-components/src/components/DocumentPreview/DocumentPreview.tsx

```diff
@@ -82,12 +84,17 @@ const DocumentPreview: FC<Props> = ({
     async function fetchFile() {
       const hasFile = await documentProvider!.provides(document!);
       if (hasFile) {
-        setProvidedFile(await documentProvider?.get(document!));
+        const file = await documentProvider?.get(document!);
+        if (file?.['__type'] === 'pdf') {
+          setProvidedFile((file as PDFSourceDocument).source);
+        } else {
+          setProvidedFile(undefined);
+        }
       }
     }
 
     // `file` takes precedence over a file provided by `documentProvider`
-    if (file) {
+    if (typeof file === 'string') {
```

Comment:
We should expand the `file` prop of `DocumentPreview` to also accept `PDFSource`. That way, a consumer of `DocumentPreview` can make use of that (for example, by simply passing in a URL instead of first donwloading all the data).